docs/backend/quantization.ipynb

```diff
@@ -0,0 +1,281 @@
+      "Loading safetensors checkpoint shards:   0% Completed | 0/4 [00:00<?, ?it/s]\n",
+      "Loading safetensors checkpoint shards:  25% Completed | 1/4 [00:00<00:01,  1.58it/s]\n",
+      "Loading safetensors checkpoint shards:  50% Completed | 2/4 [00:01<00:01,  1.30it/s]\n",
+      "Loading safetensors checkpoint shards:  75% Completed | 3/4 [00:02<00:00,  1.23it/s]\n",
+      "Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:02<00:00,  1.61it/s]\n",
+      "Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:02<00:00,  1.49it/s]\n",
+      "\n",
+      "[2024-12-27 05:23:49 TP0] Load weight end. type=LlamaForCausalLM, dtype=torch.bfloat16, avail mem=69.68 GB\n",
+      "[2024-12-27 05:23:49 TP0] Memory pool end. avail mem=8.31 GB\n",
+      "[2024-12-27 05:23:50 TP0] Capture cuda graph begin. This can take up to several minutes.\n",
+      "100%|██████████| 23/23 [00:08<00:00,  2.60it/s]\n",
+      "[2024-12-27 05:23:59 TP0] Capture cuda graph end. Time elapsed: 8.85 s\n",
+      "[2024-12-27 05:23:59 TP0] max_total_num_tokens=493601, max_prefill_tokens=16384, max_running_requests=2049, context_len=131072\n",
+      "[2024-12-27 05:23:59] INFO:     Started server process [1369]\n",
+      "[2024-12-27 05:23:59] INFO:     Waiting for application startup.\n",
+      "[2024-12-27 05:23:59] INFO:     Application startup complete.\n",
+      "[2024-12-27 05:23:59] ERROR:    [Errno 98] error while attempting to bind on address ('0.0.0.0', 30000): address already in use\n",
+      "[2024-12-27 05:23:59] INFO:     Waiting for application shutdown.\n",
+      "[2024-12-27 05:23:59] INFO:     Application shutdown complete.\n",
+      "[2024-12-27 05:24:00] The server is fired up and ready to roll!\n"
+     ]
+    }
+   ],
+   "source": [
+    "from sglang.utils import (\n",
+    "    execute_shell_command,\n",
+    "    wait_for_server,\n",
+    "    terminate_process,\n",
+    "    print_highlight,\n",
+    ")\n",
+    "\n",
+    "server_process = execute_shell_command(\n",
+    "    \"\"\"\n",
+    "python3 -m sglang.launch_server --model-path meta-llama/Meta-Llama-3.1-8B-Instruct --quantization fp8 --port 30000 --host 0.0.0.0\n",
+    "\"\"\"\n",
+    ")\n",
+    "\n",
+    "wait_for_server(\"http://localhost:30000\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our team is working on supporting more quantization methods. We will soon support other quantization methods including but not limited to `[\"awq\", \"gptq\", \"marlin\", \"gptq_marlin\", \"awq_marlin\", \"bitsandbytes\", \"gguf\"]`\n",
+    "\n",
+    "We also support quantization methods based on [torchao](https://github.com/pytorch/ao). You can simply specify `--torchao-config` in the command line to support this feature. For example, if you want to enable `int4wo-128` for model `meta-llama/Meta-Llama-3.1-8B-Instruct`, you can launch the server with the following command:\n",
```

Comment:
Give link to torchao. Also, are you sure that this kind of online quantization is recommended?

Reply:
explain when to use torchao and how to do the right things.